tests/writer_test.py

```diff
@@ -0,0 +1,100 @@
+# -*- coding: utf-8 -*-
+"""Tests for the artifact definitions readers."""
+
+import io
+import os
+import sys
+import unittest
+import yaml
+import json
+
+from artifacts import errors
+from artifacts import reader
+from artifacts import writer
+
+
+class ArtifactAsDictTest(unittest.TestCase):
```

Comment:
I set this up to test that AsDict gets you the same dict as reading the artifact using yaml alone, then I test to make sure all the defined artifacts can be converted without raising.

The Writer tests are ensuring that, while using AsDict, they can convert to string output and read that string back in without losing anything.